### PR TITLE
Update code to latest APIs

### PR DESCRIPTION
**Pull request by rollandf**
A few updates as some interfaces changed.

- Commented out `serializer.DirectCodecFactory`, as it does not exists anymore (Any aother API to use?)
- Added `UpdateOptions` to `ResourceInterface.Update`
- Added `UpdateOptions` in `UpdateStatus`
- Added `CreateOptions` in `Create`
- Updated `Deployment` API version in examples
- Remove use of `coreos-inc/tectonic-operators`
- Added go.mod file
- Use `ProcessCollectorOpts` in `NewProcessCollector`
- Use `jsonnet.Contents` in `Import`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ This section will layout a few examples gradually going from very simple to more
 Firstly install the `locutus`:
 
 ```
-go get github.com:brancz/locutus
+go get github.com:rollandf/locutus
```
Comment (brancz) [inline README.md]: can you revert this back?

File: README.md [hunk 2]
```diff
@@ -120,7 +120,7 @@ kubectl create -f example/jsonnet-with-crd-as-config/grafana.yaml
 
 If the built in functionality is not sufficient, additional renderers, triggers and rollout actions can be injected with only few lines of code.
 
-TODO(brancz): explain how everything is a library, and main.go can be viewed as just a sample implementation.
+TODO(rollandf): explain how everything is a library, and main.go can be viewed as just a sample implementation.
```
Comment (brancz) [inline README.md]: I'm happy if you want to do it, but not necessary :)

File: client/client.go [hunk 3]
```diff
@@ -122,7 +121,7 @@ func setConfigDefaults(groupVersion string, config *rest.Config) error {
 	if config.GroupVersion.Group == "" && config.GroupVersion.Version == "v1" {
 		config.APIPath = "/api"
 	}
-	config.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: scheme.Codecs}
+	//config.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: scheme.Codecs}
```
Comment (brancz) [inline client/client.go]: I assume we can then just delete this line?

File: config/config.go [hunk 4]
```diff
@@ -3,8 +3,8 @@ package config
 import (
 	"io/ioutil"
 
-	"github.com/brancz/locutus/rollout"
-	"github.com/brancz/locutus/trigger/types"
+	"github.com/rollandf/locutus/rollout"
```
Comment (brancz) [inline config/config.go]: you want to have the repo in your gopath but under my user, then you don't need thi

File: README.md (hunk 1 above)
Comment (rollandf) [reply README.md]: Yes, sure. I messed up with the PR.
I will send a fix soon.

File: config/config.go (hunk 4 above)
Comment (rollandf) [reply config/config.go]: Done

File: README.md (hunk 2 above)
Comment (rollandf) [reply README.md]: Reverted

File: main.go [hunk 5]
```diff
@@ -128,11 +127,6 @@ func Main() int {
 
 	mux := http.NewServeMux()
 	mux.Handle("/metrics", promhttp.InstrumentMetricHandler(reg, promhttp.HandlerFor(reg, promhttp.HandlerOpts{})))
-	mux.Handle("/debug/pprof/", http.HandlerFunc(pprof.Index))
```
Comment (brancz) [inline main.go]: just use the stdlib pprof library, the tectonic operators import was probably caused by auto-imports accidentally

**Comment by brancz** [comment]
Thanks! lgtm :+1:

**Pull request merged**